### ISSUE TITLE
Rule proposal: `vue/prefer-separate-static-class`

### ISSUE THREAD

**Issue by FloEdelmann** [vuejs/eslint-plugin-vue#1726]
**Please describe what the rule should do:**
Report static classes in bound class attributes.

**What category should the rule belong to?**
[x] Enforces code style (layout)
[ ] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

```vue
<!-- FAIL -->
<div v-bind:class="'static-class'" />
<div :class="'static-class'" />
<div :class="`static-class`" />
<div :class='"static-class"' />
<div :class="['static-class']" />
<div :class="[`static-class`]" />
<div :class='["static-class"]' />
<div :class="{'static-class': true}" />
<div :class="{foo: true}" />
<div :class='{"static-class": true}' />
<div :class="{['static-class']: true}" />
<div :class="{[`static-class`]: true}" />
<div :class="[{'static-class': true}]" />

<div :class="['static-class', dynamicClass]" />
<div :class="{'static-class': true, 'dynamic-class': foo}" />
```

```vue
<!-- PASS -->
<div class="static-class" />
<div :class="dynamicClass" />
<div class="static-class" :class="dynamicClass" />
<div class="static-class" :class="[dynamicClass]" />
<div class="static-class" :class="{'dynamic-class': foo}" />>
```

**Comment by ota-meshi** [comment]
I like that rule :+1:
